sync/awsdatabricks.py

```diff
@@ -799,6 +852,89 @@ def _wait_for_cluster_termination(
     return Response(error=DatabricksAPIError(**cluster))
 
 
+def monitor_cluster(cluster_id: str, polling_period: int = 30) -> None:
+    cluster = get_default_client().get_cluster(cluster_id)
+    cluster_log_dest = _cluster_log_destination(cluster)
+    if cluster_log_dest:
+        (_, bucket, base_prefix) = cluster_log_dest
+
+        aws_region_name = DB_CONFIG.aws_region_name
+        ec2 = boto.client("ec2", region_name=aws_region_name)
+        s3 = boto.client("s3")
+
+        # If the event log destination is just a *bucket* without any sub-path, then we don't want to include
+        #  a leading `/` in our Prefix (which will make it so that we never actually find the event log), so
+        #  we make sure to re-strip our final Prefix
+        file_key = f"{base_prefix}/sync_data/cluster_instances.json".strip("/")
+        previous_instances = None
+
+        while True:
+            try:
+                instances = ec2.describe_instances(
+                    Filters=[
+                        {"Name": "tag:Vendor", "Values": ["Databricks"]},
+                        {"Name": "tag:ClusterId", "Values": [cluster_id]},
+                        # {'Name': 'tag:JobId', 'Values': []}
+                    ]
+                )
+
+                if previous_instances:
+                    logger.info("Merging instances....")
+                    new_instances = [res for res in instances["Reservations"]]
+                    new_instance_id_to_reservation = dict(
+                        zip(
+                            [res["Instances"][0]["InstanceId"] for res in new_instances],
+                            new_instances,
+                        )
+                    )
+
+                    old_instances = [res for res in previous_instances["Reservations"]]
+                    old_instance_id_to_reservation = dict(
+                        zip(
+                            [res["Instances"][0]["InstanceId"] for res in old_instances],
+                            old_instances,
+                        )
+                    )
+
+                    old_instance_ids = set(old_instance_id_to_reservation.keys())
+                    new_instance_ids = set(new_instance_id_to_reservation.keys())
```

Comment:
I don't think `.keys` is needed

Reply:
oh, right you are! Interesting that Python just gives you the keys... what's the point of `.keys()`, then? Indeed, according to the docs, `.keys()` kind of seems superfluous in modern Python -

> iter(d)
      Return an iterator over the keys of the dictionary. This is a shortcut for iter(d.keys()).

https://docs.python.org/3/library/stdtypes.html#typesmapping